### PR TITLE
Add documentation for the elastic-agent diagnostics pprof commands

### PR DESCRIPTION
**Pull request by michel-laterman**
Add command reference docs for gathering pprof data with the
elastic-agent diagnostics collect and diagnostics pprof commands. Add
secrtion to troubleshooting guide for gathering pprof data into archive.

### CODE REVIEW

**Review by dedemorton — COMMENTED** [review]
A few comment. Otherwise, looks great! Thanks for adding this.

File: docs/en/ingest-management/commands.asciidoc [hunk 1]
```diff
@@ -60,7 +74,30 @@ Otherwise specify `json`, `yaml`, or `human` (the default).
...
+The length for the `trace` and `profile` data that's collected when gathering `pprof` data.
+Must be a string that can be parsed as a `time.Duration` value (default: `30s`).
+
+`--timeout <string>`::
+The timeout value for the `diagnostics collect` or `diagnostics pprof` command.
+Should be longer then `pprof-duration` as the command needs to gather and process the data.
+Must be a string that can be parsed as a `time.Duration` value (default: `30s + pprof-duration`).
+
+`--pprof-type <string>`::
+The pprof data to gather, one of: `allocs`, `block`, `cmdline`, `goroutine`, `heap`, `mutex`, `profile`, `threadcreate`, or `trace` (default: `profile`).
```
[the remark this answers is not in the document]
Comment (michel-laterman) [reply docs/en/ingest-management/commands.asciidoc:93]: yes, these types are part of `pprof`, not something we're implementing

**Comment by EricDavisX** [comment]
@dikshachauhan-qasource pinging you in the Docs PR - we should review docs as well as test out the diagnostics as part of it on the 3 OSes, with Endpoint and OSQuery enabled as test subjects.

@michel-laterman it will collect Endpoint, yes?  And logs from any other sub-process (like OSQuery) or just Beats (like Filebeat/Metricbeat)?

Is there any environment where it is known not to work, like when deployed as a Docker container?

So very cool - so glad we have docs, thanks for working it.

**Comment by michel-laterman** [comment]
@EricDavisX I know it will collect all *beats. I don't think it will collect from `endpoint` as it logs under a different path (and i don't know if the agent is aware of endpoint's install path). I have not tested if it interacts with other processes like OSQuery

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated elastic-agent diagnostics pprof command on below agents:
- Windows
- Linux
- MAC 

We added below integrations to these agents:
- System
- Endpoint Security
- OS Query

**Output for Windows:**


**Output for MAC:**




**Output for Linux:**


**Observations on running `elastic-agent diagnostics pprof`:**
- Data is available for filebeat
- Data is available for metricbeat
- Data is available for OS-Query module.
- Name: Endpoint Security is there for all agents but it shows ` error: Get "http://npipe/": open \\.\pipe\default-endpoint-security: The system cannot find the file specified.`.

We have also tested this with `sudo elastic-agent diagnostics collect` command, referred from [guide](https://www.elastic.co/guide/en/fleet/7.16/fleet-troubleshooting.html#_collect_elastic_agent_diagnostics_bundle):
We get the desired zip file for logs on all 3 OS's:
**MAC:**
[elastic-agent-diagnostics-2021-12-07T08-53-43Z-00-MAC.zip](https://github.com/elastic/observability-docs/files/7667398/elastic-agent-diagnostics-2021-12-07T08-53-43Z-00-MAC.zip)

**Windows:**
[elastic-agent-diagnostics-2021-12-07T08-07-46Z-00-Windows.zip](https://github.com/elastic/observability-docs/files/7667399/elastic-agent-diagnostics-2021-12-07T08-07-46Z-00-Windows.zip)

**Linux:**
[elastic-agent-diagnostics-2021-12-07T08-46-44Z-00-Linux.zip](https://github.com/elastic/observability-docs/files/7667397/elastic-agent-diagnostics-2021-12-07T08-46-44Z-00-Linux.zip)

> I don't think it will collect from endpoint as it logs under a different path

As also confirmed by @michel-laterman we got same output:
These zip files do contains logs for filebeat, metricbeat and os-querybeat but no logs for Endpoint Security.

Further, we will be testing docs whenever this ticket will be merged.

Please let us know if we are missing anything.
Thanks

**Review by dedemorton — APPROVED** [review]
LGTM.

**Comment by dedemorton** [comment]
@michel-laterman Not sure if you need to make any more content changes, but what's here looks good to me. I've added the backport labels, but can't remember if this is valid for 7.16. Just remove that backport label if it's not. Thanks!

**Pull request merged**